src/Airship/Internal/Decision.hs

```diff
@@ -559,12 +559,17 @@ l13 r@Resource{..} = do
         Nothing ->
             m16 r
 
-l07 r = do
+l07 r@Resource{..} = do
     trace "l07"
     req <- lift request
-    if requestMethod req == HTTP.methodPost
-        then m07 r
-        else lift $ halt HTTP.status404
+    if requestMethod req == HTTP.methodPost then m07 r
+    else do
+        m <- _contentType <$> get
+        when (isJust m) $ do
+            let (Just (ct, body)) = m
+            lift body >>= lift . putResponseBody
+            lift $ addResponseHeader ("Content-Type", renderHeader ct)
```

Comment:
Good questions. I used `contentTypesProvided` because it does match the expectation put forth by the example and also other code that we have written. Another nice thing is that it allows for different responses based on the `Accept` header in the request. It could be that having a full blown `Resource` for handling 404 is overkill. Perhaps instead just accepting some function that returns a `ResponseBody` would be sufficient.

The erlang version provides a default 404 error response with type `text/html` and a default body, but it also allows for the default error handler module to be overridden with a custom error handler that does whatever you like. I do wonder if taking a similar approach and having a default 404 response similar to what is in the Basic example and making the 404 resource optional would be a better. 

Another unfortunate situation that makes me think the 404 customization should be optional is you can't leave `contentTypesProvided` as an empty list in the 404 resource because that results in requests to undefined routes returning a 406. So you are forced into some boilerplate you may care nothing about. My change to avoid using the unsafe `head` as in `o18` was intentional and an attempt at a default response composed of just a 404 status code with no response body, but the processing at `c04` mucks that up and 
you get a 406.